### PR TITLE
Add support for OpenShift 4.7

### PR DESCRIPTION
**Pull request by csrwng**
Modifies the way that the kube apiserver is initialized by adding a
sidecar that applies required resources to it before the API server
becomes available.
Deploys a separate apiserver for OpenShift OAuth and User resources.
This requires that a new service IP be allocated and referenced from
inside the cluster.
Makes significant changes to the Kube API server config.

### CODE REVIEW

**Comment by csrwng** [comment]
## Summary of Changes for 4.7

### Kube API Server (KAS)

- Much of the configuration for KAS has moved to the apiServerArguments field of the config instead of separate sections in the config.yaml
  This means that a default audit policy can no longer be embedded in the config.yaml, it needs to be provided in a separate file. This PR adds a new configmap (`apiserver-default-audit-cm`) to contain the default audit policy. When a custom one is specified,  the custom one will be mounted in a separate configmap.
- The KAS no longer becomes available without bootstrap manifests being applied to it. The way bootstrap manifests were applied previously is that the `manifest-bootstrapper` pod would extract them and apply them to the API server. This is no longer possible because the manifest-bootstrapper pod uses the local service endpoint which relies on the API server being available. The portion of manifests that are required for the KAS to become available are now initialized via a sidecar container on the KAS pod itself.
- Because there is a new sidecar that needs to talk to the KAS (to bootstrap manifests), **a new kubeconfig is needed that points to `localhost` and is mounted to the KAS pod. The name for the secret is `localhost-admin-kubeconfig` with a single key of `kubeconfig`**

### Kube Controller Manager (KCM)
- Much of the configuration file for the KCM is now ignored. The configuration has moved to arguments on the KCM deployment.

### Openshift API Server / OpenShift OAuth API Server
- The OpenShift API server has been split into 2 servers:
   - openshift-api-server
   - openshift-oauth-api-server

- The new OAuth API Server will need to be wired in the same way as OpenShift API Server. By that I mean that we will need to add a proxy service within the child cluster with endpoints that point to the service endpoint on the tugboat cluster. 
- **A service will need to be pre-created for the OAuth API Server and its cluster IP passed in `oauthAPIClusterAPI` to the ibm-roks CLI**
- **A secret containing PKI named `openshift-oauth-apiserver` will need to be pre-created.** For what the secret looks like, see `assets/oauth-apiserver/oauth-apiserver-secret`
- Fields have been added to the cluster.yaml to provide Oauth API server resource requests and security context information.

File: assets/kube-controller-manager/kube-controller-manager-deployment.yaml [hunk 1]
```diff
@@ -70,6 +70,38 @@ spec:
         args:
         - "--openshift-config=/etc/kubernetes/cmconfig/config.yaml"
         - "--kubeconfig=/etc/kubernetes/secret/kubeconfig"
+        - "--authentication-kubeconfig=/etc/kubernetes/secret/kubeconfig"
```
Comment (rtheis) [inline assets/kube-controller-manager/kube-controller-manager-deployment.yaml:73]: Should these be removed from `assets/kube-controller-manager/config.yaml` as well?

File: assets/kube-apiserver/config.yaml [hunk 2]
```diff
@@ -17,120 +17,195 @@ admission:
...
         - {{ .PodCIDR }}
         - {{ .ServiceCIDR }}
-aggregatorConfig:
-  proxyClientInfo:
-    certFile: "/etc/kubernetes/secret/proxy-client.crt"
-    keyFile: "/etc/kubernetes/secret/proxy-client.key"
 apiServerArguments:
+  advertise-address:
+  - "{{ .ExternalAPIIPAddress }}"
+  allow-privileged:
```
Comment (rtheis) [inline assets/kube-apiserver/config.yaml:23]: I was struggling to find the previous setting for `allow-privileged` along with the following: `anonymous-auth`, `authorization-mode`, `enable-admission-plugins`, `enable-logs-handler`, `enable-swagger-ui`, `endpoint-reconciler-type`, `etcd-prefix`, `event-ttl`, `goaway-chance`, `insecure-port`, `kubelet-https`, `kubelet-read-only-port`, `kubernetes-service-node-port`, `max-mutating-requests-inflight`, and `service-account-lookup`.  Are these the previous defaults?

**Review by rtheis — APPROVED** [review]
Thanks Cesar.  I have several comments but am fine moving forward with this as-is so we can unblock 4.7 and 4.8 work.

**Pull request merged**